### ISSUE TITLE
Parsing bug (unable to parse valid XML)

### ISSUE THREAD

**Issue by xxxserxxx** [sibprogrammer/xq#21]
xq 1.1.1 fails to parse valid XML, and I suspect this is a regression. I found this after an upgrade.

I use xq in a script to parse RSS from https://mastodon.social/@flameReactor.rss . After upgrading to 1.1.1, xq started generating errors for the RSS.  Previous versions had been working.

I narrowed it down to a small subset of the XML that generates the error. Given the input:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:webfeeds="http://webfeeds.org/rss/1.0" xmlns:media="http://search.yahoo.com/mrss/">
  <channel>
    <title>Flame Reactor</title>
    <description>Public posts from @flameReactor@mastodon.social</description>
    <link>https://mastodon.social/@flameReactor</link>
    <image>
      <url>https://files.mastodon.social/accounts/avatars/108/494/405/727/932/593/original/73c676559a899098.png</url>
      <title>Flame Reactor</title>
      <link>https://mastodon.social/@flameReactor</link>
    </image>
  </channel>
</rss>
```

xq generates an unexpected syntax error at the first `<link>` element.

```bash
$ xq xml2
XML syntax error on line 6: unexpected end element </link>
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:webfeeds="http://webfeeds.org/rss/1.0" xmlns:media="http://search.yahoo.com/mrss/">
  <channel>
    <title>Flame Reactor</title>
    <description>Public posts from @flameReactor@mastodon.social</description>
    <link/>https://mastodon.social/@flameReactor</channel>
</rss>
```

xmllint confirms the input is valid xml:

```bash
$ xmllint xml2 > /dev/null && echo VALID
VALID
```

Dist: Artix, 6.0.12-artix1-1
xq: xq version 1.1.1 (2022-12-19, f0c2fb2fb0dcc90e9422a219ffdf437113777388)

**Comment by sibprogrammer** [comment]
Thank you for your report. The regression was introduced in 4a1efbc9c39446b3149e0305162726f5343b4542

**Comment by namespacebrian** [comment]
I was very close to re-reporting this issue, but thought to check closed issues before opening a new one.  Would it make sense to keep issues open until the fix is rolled into a new release?  Just a thought...

**Comment by sibprogrammer** [comment]
@namespacebrian It usually works that way. The issue is closed as soon as the fix is committed to the master branch. It may produce duplicate bug reports, but it helps to keep track of closed issues (I even do it using a commit message with the corresponding syntax).

**Comment by sibprogrammer** [comment]
Oh, sorry, the same issue here as well. I reported it separately. See #25

**Comment by xxxserxxx** [comment]
All fixed with 1.1.3, thank you!
